mw-webapp/src/component/tag/Tag.tsx

```diff
@@ -47,25 +68,34 @@ interface TagProps {
  * WayCard tag component
  */
 export const Tag = (props: TagProps) => {
-  return (
-    <button className={styles.tagContainer}>
-      <span
-        className={clsx(styles.cross, !props.isDeletable && styles.hiddenCross)}
-        onClick={() => {
-          props.isDeletable && props?.onDelete && props.onDelete(props.tagName);
-        }}
-        data-cy={props.cy?.dataCyCross}
-      >
-        X
-      </span>
+  return props.type === TagType.BASIC_TAG ?
+    (
+      <button className={styles.tagContainer}>
+        <span
+          className={clsx(styles.cross, !props.isDeletable && styles.hiddenCross)}
+          onClick={() => {
+            props.isDeletable && props?.onDelete && props.onDelete(props.tagName);
+          }}
+          data-cy={props.cy?.dataCyCross}
+        >
+          X
+        </span>
 
+        <span
+          className={styles.tag}
+          data-cy={props.cy?.dataCyTag}
+        >
+          {props.tagName}
+        </span>
+      </button>
+    )
+    :
+    (
       <span
-        className={styles.tag}
+        className={styles.wayTag}
         data-cy={props.cy?.dataCyTag}
       >
         {props.tagName}
       </span>
-    </button>
-  );
```

Comment:
Looking as duplicate code 94-99 lines and 84-89. Let's try to get rid of duplicate

Reply:
Done